### ISSUE TITLE
3d_armor doesn't enchant properly

### ISSUE THREAD

**Issue by aerozoic** [minetest-mods/xdecor#81]
I have a fresh install of 3d_armor and xdecor. I've not made any modifications to either mod. Every single piece of armor of every variety, when enchanted, shows a level of 1.8 on the HUD. I've also done a kill test with diamond armor. Wearing the full set, it took 21 clicks with a steel sword to kill while the enchanted diamond armor only took 10 clicks.
:large_orange_diamond:

**Comment by kilbith** [comment]
@stujones11 Something that could have broke my mod in 3d_armor recently?

**Comment by jastevenson303** [comment]
cessna151 maybe it makes a difference if you're using clothing, multiskin, and 3d_armor multiskin branch?

**Comment by cessna151** [comment]
Yes i am using 3d_armor multiskin branch. If it helps this is the repo for my server: https://github.com/UGXaero/UGXrealms
:large_orange_diamond:

**Comment by mgl512** [comment]
> Something that could have broke my mod in 3d_armor recently?

It's been broken for ages by 3d_armor's new api. We fixed it in [our rather heavily updated fork of xdecor](https://github.com/MT-Eurythmia/xdecor) but I didn't bother to report. You may want to check this commit: https://github.com/MT-Eurythmia/xdecor/commit/cff1ba48e11bb34af3910d6fbce31c46b3178709. I don't know if it was broken even more after that. Looked ok last month. 

I remember that the enchanment table requires armor textures names it doesn't provide and that it doesn't even use them if you provide them. It uses the ones from 3d_armor.

**Comment by kilbith** [comment]
So now it's clear we can't rely safely on 3d_armor's API from external mods.
Thanks for such maintenance, @stujones11 :-1: 

Someone needs to import the patches from @mgl512. And please mind to backport fixes in upstream, it's very important.

**Comment by kilbith** [comment]
Problem radically solved: https://github.com/minetest-mods/xdecor/commit/85b023cfe7d721dae8602bca1d973b7eb8c2cf70

I don't give a crap about unstable APIs causing deal-breaking regressions on other mods.

**Comment by stujones11** [comment]
Well that should not have been necessary, I am sure there was much less radical solution had I know about this issue (sorry I missed your 'pings'). I did make substantial re-work of the mod a while back but did my best to maintain backwards compatibility with other mods, obviously I must have missed something though I am sure it would have been a simple enough fix.

**Comment by aerozoic** [comment]
@sofar Would you be willing to reopen this issue and work with @stujones11 to resolve it properly?

Thanks
🔶

**Comment by stujones11** [comment]
Unfortunately, it looks like it was broken by adding support for non-fleshy damage groups so there is no easy fix form my end. To be fair though, this has nothing to do with unstable api's as these groups were not previously documented and the method used for enchantment was far from ideal.

There is, however, an open feature request for supporting `ItemStackMetaRef` which should make things like this much easier https://github.com/stujones11/minetest-3d_armor/issues/114 I will make a PR to re-add 3d_armor support once this is done.

In the meantime, @mgl512 has already provided a patch for this  https://github.com/minetest-mods/xdecor/issues/81#issuecomment-325926337 This could have simply been cherry-picked had the support not been so 'radically' removed.

**Comment by sofar** [comment]
I'll take a patch that reverts 85b023c and adds the patch from @gml512 on top of it. Please make a PR.

**Comment by stujones11** [comment]
@sofar thanks but if I am making a PR it will be(hopefully) a cleaner implementation when I add support for `ItemStackMeta`. However, I have no objection to anyone else doing this. Since this is now really just a request so you can, at least, remove the 'bug' label :)

**Comment by aerozoic** [comment]
If this isn't gonna be fixed, would somebody please explain how i can fix this on my server? I'm not a coder. Thanks
:large_orange_diamond:

**Comment by sofar** [comment]
Indentation needs to remain using tabs and not spaces. Please change the spaces to tabs again.
